theano/compile/builders.py

```diff
@@ -263,48 +274,50 @@ def _recompute_grad_op(self):
 
         if isinstance(grad_op, OpFromGraph):
             self._grad_op_is_cached = True
-            self._grad_op_overrides_l = [None] * len(self.local_inputs)
+            self._grad_op_overrides_l = [None] * inp_len
             return
 
         output_grads = [out_t() for out_t in self.output_types]
+        fn_grad = partial(
+            theano.gradient.grad,
+            cost=None,
+            disconnected_inputs='ignore',
+            return_disconnected='Disconnected',
+            null_gradients='return',
```

Comment:
@lamblin I think the usage of return_disconnected and null_gradients here is well done, but a double check would be great. Scan.grad use null_gradients='return and return_disconnected='None'